app/src/main/java/com/pavesid/androidacademy/ui/details/DetailsFragment.kt

```diff
@@ -203,34 +250,64 @@ class MoviesDetailsFragment :
         binding.detailsCastRecycler.children.forEach {
             it.startAnimation(rotate)
         }
+
+        binding.detailsCrewRecycler.children.forEach {
+            it.startAnimation(rotate)
+        }
     }
 
     private fun initView() {
-        initMovie()
+        if (currentMovie.backdrop.isNotBlank()) {
+            binding.detailsOrig.load(currentMovie.backdrop.toRightUrl()) {
+                crossfade(true)
+            }
+        } else {
+            binding.detailsOrig.load(BACKDROP_PLACEHOLDER) {
+                crossfade(true)
+            }
+        }
         binding.apply {
             detailsStorylineTitle.setShaderForGradient()
-            detailsHeading.setShaderForGradient()
+            detailsCastHeading.setShaderForGradient()
             detailsCastRecycler.apply {
-                layoutManager = this@MoviesDetailsFragment.layoutManager
+                layoutManager = this@DetailsFragment.layoutManagerCast
                 adapter = castAdapter
-                addItemDecoration(castItemDecoration)
+                addItemDecoration(customItemDecoration)
             }
+            detailsCrewRecycler.apply {
+                layoutManager = this@DetailsFragment.layoutManagerCrew
+                adapter = crewAdapter
+                addItemDecoration(customItemDecoration)
+            }
+
+            detailsStoryline.text = currentMovie.overview
+            collapsingToolbar.title = currentMovie.title
+            detailsTag.text = currentMovie.genres.joinToString { it.name }
+            detailsRating.rating = currentMovie.ratings / 2
+            detailsReviews.text =
+                resources.getQuantityString(
+                    R.plurals.review,
+                    currentMovie.numberOfRatings,
+                    currentMovie.numberOfRatings
+                )
+            detailsRectangle.text = getString(R.string.pg, currentMovie.minimumAge)
         }
     }
 
     companion object {
-        private const val PARAM_PARCELABLE = "ParcelableElement"
+        private const val PARAM_MOVIE = "movie"
+        private const val BACKDROP_PLACEHOLDER = "https://hollywoodsuite.ca/wp-content/uploads/poster/hws-placeholder.jpg"
 
         @JvmStatic
         fun newInstance(
-            parcelable: Parcelable,
+            id: String,
```

Comment:
1. Why not Parcelable?
2. Argument name should be Movie instead of Id?

Reply:
Why should I use Parcelable if the object is Serializable